Cython/Compiler/Symtab.py

```diff
@@ -2658,6 +2651,13 @@ def adapt(cname):
             if base_entry.utility_code:
                 entry.utility_code = base_entry.utility_code
 
+        # If the class defined in a pxd, specific entries have not been added.
+        # Ensure now that the parent (base) scope has specific entries
+        # Iterate over a copy as get_all_specialized_function_types() will mutate
+        for base_entry in base_scope.cfunc_entries[:]:
+            if base_entry.type.is_fused:
+                base_entry.type.get_all_specialized_function_types()
```

Comment:
I moved specialisation of **after** creating new cfunction entries to ensure that the entry will have name of generic function instead of secialized one. This will fix lookup issue mentioned here: https://github.com/cython/cython/issues/3607#issuecomment-1872322772

Reply:
Unfortunately, this fixes the Issue but breaks existing testcases... More investigation is needed...

Reply:
OK, the issue is that fixing this bug is causing that following tests start failing:

https://github.com/cython/cython/blob/c22c4e2d14d8a69b90c5ea1298d3976ae3bdbec1/tests/run/public_fused_types.srctree#L173-L189

If I am not wrong, the issue occurs on non-fused methods on master - e.g. following example fails:

```cython
```

Reply:
Yeah agreed - just linking to your issue for completeness: https://github.com/cython/cython/issues/5932